spec/index.html

```diff
@@ -153,1920 +183,1796 @@ <h2>Semantic Extensions and Entailment Regimes</h2>
+          The full semantics for typed literals is given in the next section.
+        </p>
+      </div>
+      <p class="technote">Simple interpretations are required to interpret all <a>names</a>,
+        and are therefore infinite.
+        This simplifies the exposition.
+        However, RDF can be interpreted using finite structures,
+        supporting decidable algorithms.
+        Details are given in <a href="#finite_interpretations" class="sectionRef"></a>. 
+      </p>
+      <p><em>IEXT(x)</em>, called the <dfn class="no-export lint-ignore">extension</dfn> of <em>x</em>,
+        is a set of pairs which identify the arguments for which the property is true,
+        that is, a binary relational extension.
+      </p>
+      <p>The distinction between <em>IR</em> and <em>IL</em> will become significant below when the semantics of datatypes are defined.
+        IL is allowed to be partial because some literals may fail to have a referent. 
+      </p>
+      <p class="technote">It is conventional to map a relation name to a relational extension directly.
+        This however presumes that the vocabulary is segregated into relation names and individual names,
+        and RDF makes no such assumption.
+        Moreover, RDF allows an IRI to be used as a relation name applied to itself as an argument.
+        Such self-application structures are used in RDFS, for example.
+        The use of the <em>IEXT</em> mapping to distinguish the relation as an object from its relational extension
+        accommodates both of these requirements.
+        It also provides for a notion of RDFS 'class' which can be distinguished
+        from its set-theoretic extension.
+        A similar technique is used in the ISO/IEC Common Logic standard [[?ISO24707]].
+      </p>
+      <p>The referent of a ground RDF graph in a simple interpretation <em>I</em> is then given by the following rules,
+        where the interpretation is also treated as a function from expressions (<a>names</a>, <a>triples</a> and <a>graphs</a>) to elements of the universe and truth values:
+      </p>
+      <section class="notoc">
+        <h6 class="defcap">Semantic Conditions for Ground Graphs</h6>
+        <div class="semantictable">
+          <p>If <em>E</em> is a literal, then <em>I(E) = IL(E)</em></p>
+          <span class="space"></span>
+          <p>If <em>E</em> is an IRI, then <em>I(E) = IS(E)</em></p>
+          <span class="space"></span>
+          <p>If <em>E</em> is a ground triple `s p o.`, then <em>I(E) = true</em> if<br>
+            <em>I(p)</em> is in <em>IP</em> and the pair &lt;<em>I(s)</em>, <em>I(o)</em>&gt;
+            is in <em>IEXT(I(p))</em><br>
+            otherwise <em>I(E) = false</em>.
+          </p>
+          <span class="space"></span>
+          <p>If <em>E</em> is a ground RDF graph, then <em>I(E) = false</em> if <em>I(E')</em> =
+            false for some triple <em>E'</em> in <em>E</em>, otherwise <em>I(E) = true</em>.
```

Comment:
GitHub fails to highlight my suggested changes. Please use your preferred diff tool.
```suggestion
          <p>If <em>E</em> is a ground RDF graph and <em>I(E') = false</em>
            for some triple <em>E'</em> in <em>E</em>, then
            <em>I(E) = false</em>; otherwise <em>I(E) = true</em>.
```